src/org/zaproxy/zap/extension/quickstart/pnh/ExtensionQuickStartPnH.java

```diff
@@ -0,0 +1,281 @@
+                + proxyParam.getProxyPort() + "/pnh/?" + API.API_NONCE_PARAM
+                + "=" + API.getInstance().getLongLivedNonce("/pnh/");
+    }
+
+    private ZapTextField getConfField() {
+        if (confField == null) {
+            confField = new ZapTextField();
+            confField.setEditable(false);
+            updateConfField(Model.getSingleton().getOptionsParam()
+                    .getApiParam().isEnabled());
+        }
+        return confField;
+    }
+
+    private void updateConfField(boolean apiState) {
+        if (confField == null) {
+            return;
+        }
+        // PnH URL Field has the same enable state as the API
+        confField.setEnabled(apiState);
+        if (apiState) {
+            confField.setText(getPlugNHackUrl());
+        } else {
+            confField.setText(Constant.messages
+                    .getString("quickstart.mitm.api.disabled"));
+        }
+    }
+
+    private JButton getConfButton() {
+        if (confButton == null) {
+            confButton = new JButton();
+            confButton.setText(Constant.messages
+                    .getString("quickstart.button.label.mitm"));
+            confButton.setToolTipText(Constant.messages
+                    .getString("quickstart.button.tooltip.mitm"));
+            confButton
+                    .setIcon(DisplayUtils.getScaledIcon(new ImageIcon(
+                            QuickStartPanel.class
+                                    .getResource("/org/zaproxy/zap/extension/quickstart/resources/plug.png"))));
+
+            updateConfButton(Model.getSingleton().getOptionsParam()
+                    .getApiParam().isEnabled());
+
+            confButton.addActionListener(new java.awt.event.ActionListener() {
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    DesktopUtils.openUrlInBrowser(getPlugNHackUrl());
+                }
+            });
+        }
+        return confButton;
+    }
+
+    private void updateConfButton(boolean apiState) {
+        if (confButton == null) {
+            return;
+        }
+        // PnH button has the same enable state as the API
+        confButton.setEnabled(apiState);
+        if (apiState) {
+            confButton.setToolTipText(Constant.messages
+                    .getString("quickstart.button.tooltip.mitm"));
+        } else {
+            confButton.setToolTipText(Constant.messages
+                    .getString("quickstart.mitm.api.disabled"));
+        }
+    }
+
+    public void updatePnhPanelElements(boolean apiState) {
+        updateConfButton(apiState);
+        updateConfField(apiState);
+    }
+
+    @Override
+    public void sessionAboutToChange(Session arg0) {
+        // Ignore
+    }
+
+    @Override
+    public void sessionChanged(Session arg0) {
+        // Ignore
+    }
+
+    @Override
+    public void sessionModeChanged(Mode mode) {
+    }

```

Comment:
This set of functions should also be consistent with inclusions of `// Ignore` comments